theano/scan_module/scan.py

```diff
@@ -594,7 +594,9 @@ def wrap_into_list(x):
         if init_out.get('taps', None) == [-1]:
 
             actual_arg = init_out['initial']
-            arg = safe_new(init_out['initial'])
+            if not isinstance(actual_arg, tensor.Variable):
+                actual_arg = tensor.as_tensor_variable(actual_arg)
```

Comment:
I'll have to look more in depth to check if that is actually correct.
If I recall correctly, having `actual_arg` not being an input variable has lead to trouble in the past.